test/integration/integration_test.go

```diff
@@ -181,6 +196,72 @@ func ensureTestDirEmpty() {
 	}
 }
 
+func setProxy() error {
+	go startProxy()
+	ip := getHostIP()
+	if ip == nil {
+		return errors.New("Not found any interface with IP and MAC.")
+	}
+	address := fmt.Sprintf("http://%v:3128", ip.String())
+	os.Setenv("MINISHIFT_HTTP_PROXY", address)
+	return nil
+}
+
+func unsetProxy() error {
+	os.Unsetenv("MINISHIFT_HTTP_PROXY")
+	if proxyListener != nil {
+		proxyListener.Close()
+	}
+	return nil
+}
+
+func startProxy() error {
+	var err error
+	proxyListener, err = net.Listen("tcp", ":3128")
+	if err != nil {
+		return err
+	}
+	proxy := goproxy.NewProxyHttpServer()
+	proxy.Verbose = true
+	proxyLog = bytes.NewBufferString("proxy log:\n")
+	proxy.Logger = log.New(proxyLog, "", log.Ldate)
+	http.Serve(proxyListener, proxy)
+	return nil
+}
+
+func getHostIP() net.IP {
```

Comment:
a similar function already exists in the minishift utils library. Please see if this is suitable and works for you.